SCClassLibrary/Common/Streams/Patterns.sc

```diff
@@ -101,6 +101,9 @@ Pattern : AbstractFunction {
 		var buffer, nodeID, defname;
 		var pattern = if(dur.notNil) { Pfindur(dur, this) } { this };
 
+        headerFormat = headerFormat ?? { server.recHeaderFormat };
+		sampleFormat = sampleFormat ?? { server.recSampleFormat };
```

Comment:
is there an advantage here of `??` over `?`? these operators always confuse me, trying to learn how/when is best to pick one over the other - any advice greatly appreciated!

Reply:
I think ?? evaluates the function on the right, while ? doesn't, just using that value directly. I might be wrong though.

Reply:
In `?`, the call `server.recSampleFormat` is always made, even if it isn't needed. With `??`, it is only made if `headerFormat` is nil.

Reply:
Thanks, thought it might be something like that!